### PR TITLE
[LLC} Add client team docs

### PR DESCRIPTION
**Pull request by chamons**
These are docs that mirror the [end user](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/core/Azure.Core/samples/ProtocolMethods.md) ones, but for client teams looking to start creating a client.

# Checklist

To ensure a quick review and merge, please ensure:
- [x] The PR has a understandable title and description explaining the _why_ and _what_.
- [x] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [x] The branch is recent enough to not have merge conflicts upon creation.

### CODE REVIEW

File: docs/protocol_method_client.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+1. Create a standard "high level" project [as documented](csharp_generate).
+
+2. Add the following key to the readme.md or autorest.md configuration file:
+
+```yaml
+low-level-client: true
+```
+
+3. Define the authentication in configuration
```
Comment (pakrym) [inline docs/protocol_method_client.md:13]: This is shared with non LLC. Should we extract? Or should this be in autorest repo?

**Review by pakrym — APPROVED** [review]

File: docs/protocol_method_client.md (hunk 1 above)
Comment (chamons) [reply docs/protocol_method_client.md:13]: I lean towards autorest repo. Should I file an issue?

Comment (pakrym) [reply docs/protocol_method_client.md:13]: Yep

Comment (chamons) [reply docs/protocol_method_client.md:13]: Done in https://github.com/Azure/autorest/issues/4126

Comment (chamons) [reply docs/protocol_method_client.md:13]: Turns out that these are actually autorest.csharp constructs:

```
                autoRest.GetValue<string[]?>("credential-types").GetAwaiter().GetResult() ?? Array.Empty<string>(),
                autoRest.GetValue<string[]?>("credential-scopes").GetAwaiter().GetResult() ?? Array.Empty<string>(),
                autoRest.GetValue<string?>("credential-header-name").GetAwaiter().GetResult() ?? "api-key",
```

Comment (chamons) [reply docs/protocol_method_client.md:13]: Do you want me to pull these up to a higher level document that isn't LLC specific?

Comment (pakrym) [reply docs/protocol_method_client.md:13]: I would leave it in LLC until we switch to the standardized model.

**Pull request merged**